external/js/cothority/src/byzcoin/byzcoin-rpc.ts

```diff
@@ -168,6 +193,52 @@ export default class ByzCoinRPC implements ICounterUpdater {
         return header.version;
     }
 
+    /**
+     * Returns an observable proof for an instance that will be
+     * automatically updated whenever the instance is changed.
+     * It returns a BehaviorSubject, which is a special observable that
+     * keeps the current value.
+     * It also caches all requests, so if there are two requests for the
+     * same id, no new observable will be created.
+     * It connects to `getNewBlocks` to be informed whenever a new block is
+     * created.
+     * @param id of the instance to return
+     * @throws an error if the instance does not exist
+     */
+    async proofObservable(id: InstanceID): Promise<BehaviorSubject<Proof>> {
+        const bs = this.cache.get(id);
+        if (bs !== undefined) {
+            return bs;
+        }
+
+        // Check if the db already has a version, which might be invalid,
+        // but still better than to wait for the network.
+        const idStr = id.toString("hex");
+        let dbProof: Proof | undefined = await this.db.getObject(idStr);
+        if (dbProof === undefined) {
+            dbProof = await this.getProofFromLatest(id);
+        }
+        if (!dbProof.exists(id)) {
+            throw new Error("this instance does not exist");
+        }
+
+        const bsNew = new BehaviorSubject(dbProof);
+        // Set up a pipe from the block to fetch new versions if a new block
+        // is available.
+        (await this.getNewBlocks())
```

Comment:
Please help humans understand this: explaining what each operator do and why you need them would help.

Reply:
Added comments - see also the description of the PR itself.